dockertest/subtestbase.py

```diff
@@ -155,8 +157,15 @@ def failif_not_in(needle, haystack, description=None):
             description = 'string'
         if needle in haystack:
             return
-        raise DockerTestFail("Expected string '%s' not in %s '%s'"
-                             % (needle, description, haystack))
+        expect_s = "Expected string '%s'" % needle
+        if '|' in needle:
+            needles = [n.strip() for n in needle.split('|')]
+            expect_s = "Expected strings %s" % needles
+            for subneedle in needles:
+                if subneedle.strip() in haystack:
+                    return
+        raise DockerTestFail("%s not in %s '%s'" %
+                             (expect_s, description, haystack))
```

Comment:
>How about just adding a delimiter="|" parameter then? 

The problem is, that has to be done in the code. But the place for defining a delimiter is in the .ini file. The code shouldn't (and, realistically, cannot) know what a suitable delimiter is for a given test. So we end up with something like:

    expected_stderr_delimiter = '*'
    expected_stderr = a|b * b|c * lmnop

    expected_stderr = self.config['expected_stderr']
    expected_delimiter = '|'
    if 'expected_stderr_delimiter' in self.config
         expected_delimiter = self.config['expected_stderr_delimiter']
    self.failif_etc(...delimiter=expected_delimiter)

...and having to do that in Every Single Call to self-failif_not_in()